src/ImageSharp/PixelFormats/Utils/Vector4Converters.Default.cs

```diff
@@ -87,6 +123,42 @@ public static class Default
                     dp = sp.ToScaledVector4();
                 }
             }
+
+            [MethodImpl(InliningOptions.ShortMethod)]
+            internal static void DangerousFromPremultipliedScaledVector4<TPixel>(
+                ReadOnlySpan<Vector4> sourceVectors,
+                Span<TPixel> destinationColors)
+                where TPixel : struct, IPixel<TPixel>
+            {
+                ref Vector4 sourceRef = ref MemoryMarshal.GetReference(sourceVectors);
+                ref TPixel destRef = ref MemoryMarshal.GetReference(destinationColors);
+
+                for (int i = 0; i < sourceVectors.Length; i++)
+                {
+                    ref Vector4 sp = ref Unsafe.Add(ref sourceRef, i);
+                    ref TPixel dp = ref Unsafe.Add(ref destRef, i);
+                    Vector4Utils.UnPremultiply(ref sp);
```

Comment:
Is this mixed loop as default generic implementation faster than a sequence of bulk conversions + premultiplications? Could be only checked by a benchmark.

Reply:
Logic would dictate yes but I can add a rough benchmark to test I suppose.

Reply:
CPU caches and pipelines follow a different kind of logic sometimes :)

Reply:
This is most likely OK for the default path, but we should be aware that there is some kind of individual threshold for the number of operations it is OK to put inside a loop without making it slower than doing separate bulk operations instead.